src/Stratis.Bitcoin/Base/Deployments/ActivationHeightProvider.cs

```diff
@@ -0,0 +1,79 @@
+﻿using NBitcoin;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Base.Deployments
+{
+    /// <summary>
+    /// Determines the activation height of a given deployment.
+    /// Also used to determine if a deployment was active at a given height.
+    /// </summary>
+    public class ActivationHeightProvider
+    {
+        private Network network;
+        private ChainIndexer chainIndexer;
+        private ThresholdConditionCache cache;
+        private int deployment;
+        private ChainedHeader lastPollExpiryHeightChecked;
+        private int pollExpiryActivationHeight;
+
+        public ActivationHeightProvider(Network network, ThresholdConditionCache cache, ChainIndexer chainIndexer, int deployment)
+        {
+            this.network = network;
+            this.cache = cache;
+            this.chainIndexer = chainIndexer;
+            this.deployment = deployment;
+            this.lastPollExpiryHeightChecked = null;
```

Comment:
Are the names of these two variables related to a specific deployment this provider was intended to facilitate? It seems to me they could be called `lastHeightChecked` and `activationHeight` respectively (and the resulting clashing local `lastHeightChecked` renamed)